lib/chef/provider/package/cab.rb

```diff
@@ -74,9 +74,13 @@ def remove_package(name, version)
         end
 
         def dism_command(command)
-          shellout = Mixlib::ShellOut.new("dism.exe /Online /English #{command} /NoRestart", timeout: new_resource.timeout)
           with_os_architecture(nil) do
-            shellout.run_command
+            result = shell_out("dism.exe /Online /English #{command} /NoRestart", { timeout: new_resource.timeout })
+            if result.exitstatus == -2146498530
+              raise Chef::Exceptions::Package, "The specified package is not applicable to this image." if result.inspect.include?("Error: 0x800f081e")
```

Comment:
Do we have anything else to test other than the whole result object (inc. stderr, stdout)? I'm also concerned that the word `Error:` could be localized to something else on non-English systems.

Reply:
@stuartpreston Done with the review comments, Please have a look on it.